pkg/scheduler/framework/interface.go

```diff
@@ -292,20 +291,21 @@ func (p PluginToStatus) Merge() *Status {
 
 	finalStatus := NewStatus(Success)
 	for _, s := range p {
-		if s.Code() == Error {
-			finalStatus.err = s.AsError()
-		}
 		if statusPrecedence[s.Code()] > statusPrecedence[finalStatus.code] {
 			finalStatus.code = s.Code()
 			// Same as code, we keep the most relevant failedPlugin in the returned Status.
 			finalStatus.failedPlugin = s.FailedPlugin()
 		}
 
-		for _, r := range s.reasons {
+		reasons := s.reasons
+		if finalStatus.err != nil {
+			reasons = s.Reasons()
+		}
+		for _, r := range reasons {
 			finalStatus.AppendReason(r)
 		}
+		finalStatus.err = s.err
```

Comment:
This changes the semantics of the error; `AsError` returns an error if the status is Unschedulable.

Reply:
In this pr ,we don't constuct `s.err` in `NewStatus`. So it may be fine? since we can get error by calling `s.AsError()`

Reply:
@ahg-g note that we were only calling `AsError` when the code was error.

just to make sure we are not overriding an error:

```suggestion
		if finalStatus.err == nil {
			finalStatus.err = s.err
		}
```

or it could even be the else for the conditional in line 301

although in reality once we find one error we stop doing more merges.